stablehlo/tests/transforms/stablehlo_legalize_qdq_to_quantized_op.mlir

```diff
@@ -0,0 +1,151 @@
+// RUN: stablehlo-opt %s -verify-diagnostics -split-input-file -allow-unregistered-dialect --stablehlo-legalize-qdq-to-quantized-op | FileCheck %s --check-prefixes=CHECK
+
+// -----
+
+// CHECK-LABEL @compose_quantized_abs_op
+// CHECK:      %[[abs0:.*]] = stablehlo.abs %arg0 : tensor<16x16x!quant.uniform<u8:f32, 3.400000e+01:16>>
+// CHECK-NEXT: return %[[abs0]] : tensor<16x16x!quant.uniform<u8:f32, 3.400000e+01:16>>
+func.func @compose_quantized_abs_op(%arg0: tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>) -> tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>> {
+    %0 = stablehlo.uniform_dequantize %arg0 : (tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>) -> tensor<16x16xf32>
+    %1 = stablehlo.abs %0 : tensor<16x16xf32>
+    %2 = stablehlo.uniform_quantize %1 : (tensor<16x16xf32>) -> tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>
+    func.return %2 : tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>
+}
+
+// -----
+
+// CHECK-LABEL @failure_uniform_quant_op_operand_not_defined_by_op
+// CHECK-NOT: stablehlo.abs {{.*}} : tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>
+// CHECK:     return
+func.func @failure_uniform_quant_op_operand_not_defined_by_op(%arg0: tensor<16x16xf32>) -> tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>> {
+  %1 = stablehlo.abs %arg0 : tensor<16x16xf32>
+  %2 = stablehlo.uniform_quantize %1 : (tensor<16x16xf32>) -> tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>
+  func.return %2 : tensor<16x16x!quant.uniform<ui8:f32, 34.0:16>>
+}
+
+// -----
+
+// CHECK-LABEL @failure_op_with_region
+// CHECK:         %cst = stablehlo.constant dense<2.000000e+00> : tensor<f32>
+// CHECK-NEXT:    %0 = stablehlo.uniform_quantize %arg0 : (tensor<4x5xf32>) -> tensor<4x5x!quant.uniform<u8:f32, 3.400000e+01:16>>
+// CHECK-NEXT:    %1 = stablehlo.uniform_dequantize %0 : (tensor<4x5x!quant.uniform<u8:f32, 3.400000e+01:16>>) -> tensor<4x5xf32>
+// CHECK-NEXT:    %2 = stablehlo.uniform_quantize %arg1 : (tensor<4x5xf32>) -> tensor<4x5x!quant.uniform<u8:f32, 3.400000e+01:16>>
+// CHECK-NEXT:    %3 = stablehlo.uniform_dequantize %2 : (tensor<4x5x!quant.uniform<u8:f32, 3.400000e+01:16>>) -> tensor<4x5xf32>
+// CHECK-NEXT:    %4 = "stablehlo.map"(%1, %3) <{dimensions = array<i64: 0, 1>}> ({
+// CHECK-NEXT:    ^bb0(%arg2: tensor<f32>, %arg3: tensor<f32>):
+// CHECK-NEXT:      stablehlo.return %cst : tensor<f32>
+// CHECK-NEXT:    }) : (tensor<4x5xf32>, tensor<4x5xf32>) -> tensor<4x5xf32>
+// CHECK-NEXT:    %5 = stablehlo.uniform_quantize %4 : (tensor<4x5xf32>) -> tensor<4x5x!quant.uniform<u8:f32, 3.400000e+01:16>>
+// CHECK-NEXT:    return %5 : tensor<4x5x!quant.uniform<u8:f32, 3.400000e+01:16>>
+
+func.func @failure_op_with_region(%arg0: tensor<4x5xf32>, %arg1: tensor<4x5xf32>) -> tensor<4x5x!quant.uniform<ui8:f32, 34.0:16>> {
```

Comment:
`failure_to_match_op_with_region`

Reply:
same `failure_to_match_` refix below..

Reply:
thanks, added prefix `failed_to_match` to test names.